RHDEVDOCS-3226 Logging 5.2 docs aggregated branch (#35445)
openshift/openshift-docs: logging/cluster-logging-release-notes.adoc

[id="cluster-logging-release-notes"]

> = Release notes for Red Hat OpenShift Logging 5.2

Reviewer: Just curious if you have an attributes file for Red Hat OpenShift Logging so that you don't have to type the full product name each time?

Author: Not yet. Planned for after this release.

Author: @anpingli @jcantrill @igor-karpukhin Please review and approve these release notes for 5.2.

* With this update, if you use the Fluentd forward protocol to forward log data over a TLS-encrypted connection, you can now use a password-encrypted private key file and specify the passphrase in the Cluster Log Forwarder configuration. For more information, see xref:../logging/cluster-logging-external.html#cluster-logging-collector-log-forward-fluentd_cluster-logging-external[Forwarding logs using the Fluentd forward protocol]. (link:https://issues.redhat.com/browse/LOG-1525[LOG-1525])

* This enhancement enables you to use a username and password to authenticate a log forwarding connection to an external Elasticsearch instance. For example, if you cannot use mutual TLS (mTLS) because a third-party operates the Elasticsearch instance, you can use HTTP or HTTPS and set a secret that contains the username and password. For more information see xref:../logging/cluster-logging-external.adoc#cluster-logging-collector-log-forward-es_cluster-logging-external[Forwarding logs to an external Elasticsearch instance]. (link:https://issues.redhat.com/browse/LOG-1022[LOG-1022])

> 

Author: @sasagarw @pmoogi-redhat @lukas-vlcek

* By default, the data model introduced in {product-title} 4.5 gave logs from different namespaces a single index in common. This change made it harder to see which namespaces produced the most logs.
+
The current release, OpenShift Logging 5.2, adds namespace metrics to the *Logging* dashboard in the {product-title} console. With these metrics, you can see which namespaces produce logs and how many logs each namespace produces for a given timestamp.
+
To see these metrics, open the *Administrator* perspective in the {product-title} web console, and navigate to *Monitoring* -> *Dashboards* -> *Logging/Elasticsearch*. (link:https://issues.redhat.com/browse/LOG-1680[LOG-1680])

> * The current release, OpenShift Logging 5.2, enables two new metrics: For a given timestamp or duration, you can see the total logs produced or logged by individual containers, and the total logs collected by the collector. These metrics are labeled by namespace, pod, and container name, so you can see how many logs each namespace and pod collects and produces. (link:https://issues.redhat.com/browse/LOG-1213[LOG-1213])

Author: @pmoogi-redhat  Please review and approve.

Reviewer 2: /lgtm

* By default, the data model introduced in {product-title} 4.5 gave logs from different namespaces a single index in common. This change made it harder to see which namespaces produced the most logs.
+
The current release, OpenShift Logging 5.2, enables two new metrics: the total logs produced or logged by individual containers, and the total logs collected by the collector. These metrics are labeled by namespace, pod, and container name. These labels enable you to see how many logs each namespace and pod produces for a given timestamp or duration.
+
To see these metrics, open the {product-title} console and click *Monitoring* -> *Logging*.
(link:https://issues.redhat.com/browse/LOG-1680[LOG-1680])

> * The current release, OpenShift Logging 5.2, enables two new metrics: For a given timestamp or duration, you can see the total logs produced or logged by individual containers, and the total logs collected by the collector. These metrics are labeled by namespace, pod, and container name, so you can see how many logs each namespace and pod collects and produces. (link:https://issues.redhat.com/browse/LOG-1213[LOG-1213])

Author: @pmoogi-redhat Please review and approve. Should we add something about where or how the user might be able to see these metrics?

[id="openshift-logging-5-2-0-new-features-and-enhancements"]
=== New features and enhancements

* With this update, if you use the Fluentd forward protocol to forward log data over a TLS-encrypted connection, you can now use a password-encrypted private key file and specify the passphrase in the Cluster Log Forwarder configuration. For more information, see xref:../logging/cluster-logging-external.html#cluster-logging-collector-log-forward-fluentd_cluster-logging-external[Forwarding logs using the Fluentd forward protocol]. (link:https://issues.redhat.com/browse/LOG-1525[LOG-1525])

* This enhancement enables you to use a username and password to authenticate a log forwarding connection to an external Elasticsearch instance. For example, if you cannot use mutual TLS (mTLS) because a third-party operates the Elasticsearch instance, you can use HTTP or HTTPS and set a secret that contains the username and password. For more information see xref:../logging/cluster-logging-external.adoc#cluster-logging-collector-log-forward-es_cluster-logging-external[Forwarding logs to an external Elasticsearch instance]. (link:https://issues.redhat.com/browse/LOG-1022[LOG-1022])

> * By default, the data model introduced in OpenShift 4.5 gave logs from different namespaces a single index in common. This change made it harder to see which namespaces produced the most logs. The current release, OpenShift Logging 5.2, adds namespace metrics to the *Logging* dashboard in the {product-title} console. With these metrics, you can see which namespaces produce logs and how many logs each namespace produces for a given timestamp. To see these metrics, open the {product-title} console and click *Monitoring* -> *Logging*.

Reviewer: I'd use the variable for {product-title} instead of OpenShift 4.5 here 🙂 unless it should be OpenShift Logging?

Author: Updated. Thanks.

Author: @pmoogi-redhat Is this the release note about "new metrics" you mentioned?

Reviewer 2: @rolfedh  it does n't capture the name of the new metrics that we added. We can add the below to this above description. 

 "The current release, OpenShift Logging 5.2, enables two new metrics which account for total logs produced (or logged by individual containers) and total logs collected by the collector. These metrics are labeled by namespace, pod and container names so you can see how many logs produced, collected in each namespace, in each pod for a given timestamp, duration"

Author: Thanks @pmoogi-redhat. I revised the proposed text and added it to the release note. Please review and comment or approve.

(link:https://issues.redhat.com/browse/LOG-1680[LOG-1680])

// [id="openshift-logging-5-2-0-deprecated-removed-features"]
// === Deprecated and removed features
//
// [IMPORTANT]
// ====
// {FeatureName} is a deprecated feature. Deprecated functionality is still included in {product-title} and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments.
//
// For the most recent list of major functionality that has been deprecated or removed within {product-title}, refer to the _Deprecated and removed features_ section of the {product-title} release notes.
// ====
//
// [id="openshift-logging-5-2-0-tbd"]
// ==== TBD-title
//

> // TBD-description

Reviewer: Just double checking this is supposed to be commented out? 🙂

[id="openshift-logging-5-2-0-bug-fixes"]
=== Bug fixes

* In previous releases, if you redeployed a full OpenShift Logging-managed Elasticsearch cluster, it might become stuck in an unhealthy state, with one non-data node running and all other data nodes shut down. This issue was caused by new certificates preventing the OpenShift Elasticsearch Operator from scaling down the non-data nodes of the Elasticsearch cluster. With this update, The OpenShift Elasticsearch Operator can scale all the data and non-data nodes down and back up again to load the new certificates. The OpenShift Elasticsearch Operator can reach the new nodes after they load the new certificates.
(link:https://issues.redhat.com/browse/LOG-1524[LOG-1524])

[id="openshift-logging-5-2-0-known-issues"]
=== Known issues

> * There is currently a known issue: If you forward logs to an external Elasticsearch server and change a configured value in the pipeline secret, such as the username and password, the fluentd forwarder loads the new secret but uses the old value to connect to external Elasticsearch server. This issue happens because the Red Hat OpenShift Logging Operator does not currently monitor secrets for content changes.
> 
> Workaround: If you change the secret, you can force the Fluentd pods to redeploy by entering `oc delete pod -l component=fluentd`.
> (link:https://issues.redhat.com/browse/LOG-1652[LOG-1652])

Reviewer: ```suggestion
* If you forward logs to an external Elasticsearch server and then change a configured value in the pipeline secret, such as the username and password, the fluentd forwarder loads the new secret but uses the old value to connect to an external Elasticsearch server. This issue happens because the Red Hat OpenShift Logging Operator does not currently monitor secrets for content changes. (link:https://issues.redhat.com/browse/LOG-1652[LOG-1652])
+
As a workaround, if you change the secret, you can force the Fluentd pods to redeploy by entering:
+
[source,terminal]
----
$ oc delete pod -l component=fluentd
----
```

Reviewer: Just some suggestions:
I would put the command in its own code block to draw attention to the workaround and make it easier for users to copy/paste to their terminal.

I also think the "There is currently a known issue:" may be redundant here, since the heading already mentions known issues?

Author: Thanks. Updated. I was following the conventions established by the Supplementary Style Guide but prefer your version.

[id="openshift-logging-about-this-release"]
== About this release

The following advisories are available for Red Hat OpenShift Logging versions 5.1.x and 5.2.x:

* link:https://access.redhat.com/errata/RHBA-2021:2885[RHBA-2021:2885 - Bug Fix Advisory. Openshift Logging Bug Fix Release 5.1.1]
* link:https://access.redhat.com/errata/RHBA-2021:2112[RHBA-2021:2112 - Bug Fix Advisory. OpenShift Logging Bug Fix Release 5.1.0]

[id="openshift-logging-supported-versions"]
== Supported versions

> * OpenShift Logging versions 5.1 and 5.2 run on {product-title} versions 4.7 and 4.8.

Reviewer: In the description of this PR it says the change applies for 4.8+, so I just wanted to check that this is accurate?

Author: Yes. This information is correct and appropriate for the OpenShift Logging 5.2 docs, which lands in enterprise-4.8.
The OpenShift Logging 5.1 docs, which landed in enterprise-4.7, contain a similar statement. https://docs.openshift.com/container-platform/4.7/logging/cluster-logging-release-notes.html

[id="openshift-logging-inclusive-language"]
== Making open source more inclusive

Red Hat is committed to replacing problematic language in our code, documentation, and web properties. We are beginning with these four terms: master, slave, blacklist, and whitelist. Because of the enormity of this endeavor, these changes will be implemented gradually over several upcoming releases. For more details, see link:https://www.redhat.com/en/blog/making-open-source-more-inclusive-eradicating-problematic-language[Red Hat CTO Chris Wright’s message].

// Release Notes by version

[id="cluster-logging-release-notes-5-2-0"]

> == OpenShift Logging 5.2.0

Reviewer: I think this could be it's own module as per https://github.com/openshift/openshift-docs/pull/35445/files#diff-39d0bf3cca6d5b60bc65b58d2fe0b7c791cefa2b64f8e915729dc4960b88162bR72 ?

Author: Unfortunately, because OCP still forbids xrefs in modules, I'm bringing this module content into the assembly file. This way, users can follow the xref links from the release note to the relevant documentation.

Reviewer: Makes sense! Thanks

This release includes link:https://access.redhat.com/errata/RHSA-TBD[RHSA-TBD OpenShift Logging Bug Fix Release 5.2.0].

[id="openshift-logging-5-2-0-new-features-and-enhancements"]
=== New features and enhancements

* TBD-description
(link:https://issues.redhat.com/browse/LOG-TBD[LOG-TBD])

* With this update, if you use the Fluentd forward protocol to forward log data over a TLS-encrypted connection, you can now use a password-encrypted private key file and specify the passphrase in the Cluster Log Forwarder configuration. For more information, see xref:/logging/cluster-logging-external.html#cluster-logging-collector-log-forward-fluentd_cluster-logging-external[Forwarding logs using the Fluentd forward protocol]
(link:https://issues.redhat.com/browse/LOG-1525[LOG-1525])

> * With this update, you can use a username and password to authenticate a log forwarding connection to an external Elasticsearch instance. For example, if you cannot use mutual TLS (mTLS) because a third-party operates the Elasticsearch instance, you can use HTTP or HTTPS and set a secret that contains the username and password. For more information see xref:TBD

Author: @igor-karpukhin WDYT? I'll add the cross reference link (xref) later.

> * With this update, you can use a username and password to authenticate a log forwarding connection to an external Elasticsearch instance. For example, if you cannot use mutual TLS (mTLS) because a third-party operates the Elasticsearch instance, you can use HTTP or HTTPS and set a secret that contains the username and password. For more information see xref:../logging/cluster-logging-external.adoc#cluster-logging-collector-log-forward-es_cluster-logging-external[Forwarding logs to an external Elasticsearch instance].

Reviewer: I would say you maybe don't need "with this update" at the beginning of each line. I'd maybe just put something like:
"If you use the Fluentd forward protocol to forward log data over a TLS-encrypted connection, you can now use a password-encrypted private key file and specify the passphrase in the Cluster Log Forwarder configuration."

Author: Thanks. Updated. Following the new release note guidelines in the SSG https://redhat-documentation.github.io/supplementary-style-guide/#release-notes

> * With this update, you can use a username and password to authenticate a log forwarding connection to an external Elasticsearch instance. For example, if you cannot use mutual TLS (mTLS) because a third-party operates the Elasticsearch instance, you can use HTTP or HTTPS and set a secret that contains the username and password. For more information see xref:../logging/cluster-logging-external.adoc#cluster-logging-collector-log-forward-es_cluster-logging-external[Forwarding logs to an external Elasticsearch instance].

Reviewer: ```suggestion
* You can now use a username and password to authenticate a log forwarding connection to an external Elasticsearch instance. For example, if you cannot use mutual TLS (mTLS) because a third-party operates the Elasticsearch instance, you can use HTTP or HTTPS and set a secret that contains the username and password. For more information see xref:../logging/cluster-logging-external.adoc#cluster-logging-collector-log-forward-es_cluster-logging-external[Forwarding logs to an external Elasticsearch instance].
```

Reviewer: Just some suggestions

Author: Thanks, @abrennan89. Updated.

(link:https://issues.redhat.com/browse/LOG-974[LOG-1022])

> 

Reviewer: Extra 2 blank lines seems unnecessary but presumably it's a personal preference? 🙂

[id="openshift-logging-5-2-0-deprecated-removed-features"]
=== Deprecated and removed features

Some features available in previous releases have been deprecated or removed.

> Deprecated functionality is still included in OpenShift Logging and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments.

Reviewer: There is a deprecated features module I think named `deprecated-feature.adoc` that you might want to use here instead?

Author: Thanks. These release notes are based on the release_notes/ocp-4-8-release-notes.adoc assembly, which doesn't include the modules/deprecated-feature.adoc module. It seems like we include this module to in assemblies for features that we are deprecating/removing. It is likely that I will probably remove this section from these release notes.

[id="openshift-logging-5-2-0-tbd"]
==== TBD-title

> TBD-description

Reviewer: Will this be removed or is there content TBA?

Author: This is a content template. I will remove it from the PR when it is no longer needed.

[id="openshift-logging-5-2-0-bug-fixes"]
=== Bug fixes

* TBD-description
(link:https://issues.redhat.com/browse/LOG-TBD[LOG-TBD])

> * Before this update, if you redeployed a full Elasticsearch cluster, it got stuck in an unhealthy state, with one non-data node running and all other data nodes shut down. This happened because new certificates prevented the Elasticsearch Operator from scaling down the non-data nodes of the Elasticsearch cluster. With this update, Elasticsearch Operator can scale all the data and non-data nodes down and then back up again, so they load the new certificates. The Elasticsearch Operator can reach the new nodes after they load the new certificates.
> (link:https://issues.redhat.com/browse/LOG-1524[LOG-1524])

Reviewer: ```suggestion
* In previous releases, if you redeployed a full Elasticsearch cluster, it might become stuck in an unhealthy state, with one non-data node running and all other data nodes shut down. This was caused by new certificates preventing the Elasticsearch Operator from scaling down the non-data nodes of the Elasticsearch cluster. With this update, The Elasticsearch Operator can scale all the data and non-data nodes down and then back up again, so that the nodes can load the new certificates. The Elasticsearch Operator can reach the new nodes after they load the new certificates.
(link:https://issues.redhat.com/browse/LOG-1524[LOG-1524])
```

Reviewer: Some suggestions for readability - this would also be a good place to add an attribute for the Operator name maybe?

Author: Good suggestion. I'll add an attribute for this and other operators in a later release: https://issues.redhat.com/browse/RHDEVDOCS-3238

Author: Updated. Thanks.

* Before this update, if you redeployed a full Elasticsearch cluster, it got stuck in an unhealthy state, with one non-data node running and all other data nodes shut down. This happened because new certificates prevented the Elasticsearch Operator from scaling down the non-data nodes of the Elasticsearch cluster. With this update, Elasticsearch Operator can scale all the data and non-data nodes down and then back up again, so they load the new certificates. The Elasticsearch Operator can reach the new nodes after they load the new certificates.
(link:https://issues.redhat.com/browse/LOG-1537[LOG-1537])

> 

Reviewer: Looks like a duplicate of the previous point?

Author: Removed. Thanks!
